packages/mui-material/src/Tooltip/Tooltip.js

```diff
@@ -545,7 +545,10 @@ const Tooltip = React.forwardRef(function Tooltip(inProps, ref) {
     };
   }, [handleClose, open]);
 
-  const handleRef = useForkRef(children.ref, setChildNode, ref);
+  // 'ref' is passed as prop in React 19, whereas 'ref' is directly attached to children in React 18
+  // below check is to ensure 'ref' is accessible in both cases
+  const childRef = Object.keys(children.props).includes('ref') ? children.props.ref : children.ref;
```

Comment:
True, what a bummer. Just tried locally.

Reply:
Which test/demo/code are you running to get this error @sai6855? I'm curious as, ideally we can follow this recommendation: https://github.com/radix-ui/primitives/pull/2811/files#r1601508869

Reply:
Agree https://codesandbox.io/p/sandbox/gracious-glitter-qvlxxy?file=%2Fsrc%2FTooltip.tsx%3A502%2C1 `const childRef = children.props.ref ?? children.ref;` seems to work fine.

Reply:
Can you make this sandbox public, I can't access it

Reply:
Ah god, I should default to StackBlitz. Fixed. CodeSandbox fixed a bit it's UX recently by showing the sandbox visibility setting with an icon, but I keep forgetting about it (so are people on Material UI). There should be a setting to configure a default mode in CodeSandbox.

Reply:
Above error is thrown in react 18, not 19. i just changed react version to 18.2.0 in your sandbox, can see error in console.

https://codesandbox.io/p/sandbox/quizzical-sea-qrp6xd?file=%2Fpackage.json%3A6%2C30

![image](https://github.com/mui/material-ui/assets/60743144/b2dbf17c-8574-494e-bfa0-78db24612032)

Reply:
@sai6855 sandbox is private 🙈

Reply:
@aarongarciah can you access it now

Reply:
If we want to avoid the warning while we're in React 18 we need to use the `Object.keys` hack 🙃

The other option is to make all fixes in the same PR where we bump to React 19 i.e. a _very_ big PR 😅

Reply:
Oh, I see.

````jsx
const keyAsProp = React.version.split(".")[0] >= "19";

const childRef = keyAsProp ? children.props.ref : children.ref;
```

could work but it's not pretty.

Reply:
Ok, better:

```jsx
  const childRef = children.props.propertyIsEnumerable("ref")
    ? children.props.ref
    : children.ref;
```

Reply:
If no ref is set on the element, `element.props.propertyIsEnumerable('ref')` will return `false`, making the program access `element.ref` that will trigger a warning, no?

Reply:
As I understand it the warning triggers in React 19 when accessing `children.props.ref`.